cli/src/rpc.rs

```diff
@@ -38,22 +37,51 @@ pub trait Serialization: Send + Sync + 'static {
 
 /// RPC is a basic, transport-agnostic builder for RPC methods. You can
 /// register methods to it, then call `.build()` to get a "dispatcher" type.
-pub struct RpcBuilder<S, C> {
-	context: Arc<C>,
+pub struct RpcBuilder<S> {
```

Comment:
This was split into RpcBuilder and RpcMethods. Methods were moved to a new builder so that we could `get_caller`s beforehand in order to instantiate the contact passed to methods (given to the `.methods(c)` method). A builder of builders!